cognite/client/data_classes/data_modeling/typed_instances.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from abc import ABC
+from collections.abc import Iterable
+from datetime import date, datetime
+from typing import TYPE_CHECKING, Any, ClassVar
+
+from typing_extensions import Self
+
+from cognite.client.data_classes._base import CogniteResource
+from cognite.client.data_classes.data_modeling.data_types import (
+    DirectRelationReference,
+)
+from cognite.client.data_classes.data_modeling.ids import ContainerId, EdgeId, NodeId, ViewId
+from cognite.client.data_classes.data_modeling.instances import PropertyValueWrite
+from cognite.client.utils._text import to_camel_case
+
+if TYPE_CHECKING:
+    from cognite.client import CogniteClient
+
+
+class InstanceProperty:
```

Comment:
I'd suggest this name instead
```suggestion
class PropertyOptions:
```